src/test_application/DebugObject.h

```diff
@@ -30,9 +31,6 @@
 #include <osp/UserInputHandler.h>
 #include <osp/types.h>
 
-namespace osp
-{
-
```

Comment:
The "Abstract" prefix in the class was there to match Magnum's naming convention which I somehow decided to follow for the first couple of months for some reason; this file is pretty old. Most of the new code in OSP follows the "I" standard.

The virtual destructors is a bit of a problem though. DebugObject heavily relies on having its lifetime managed by something that doesn't know its implementation. I'm not sure how this can even be fixed, or if it's even worth restructuring this class which is kind of made to be this disgusting. This is not a core component, and the use of this class is discouraged.

The purpose of the DebugObject is to be able to quickly add functionality to entities in the ECS for testing only. One of the downsides of ECS is that it takes a bit more effort to add quick and unique functions like the camera controller. An entirely new system has to be created to implement it. DebugObject along with ACompDebugObject pretty much wraps an OOP-style class into ECS.

Other than that, IDynamicSystem from `osp/Active/activetypes.h` relies on virtual destruction too, and is a topic for future discussion.

Reply:
Are you managing the lifetime of it using std::shared_ptr or std::unique_ptr? If so you can specialize the smart pointers with custom deleters that know of a virtual destructor in the hierarchy of the object that can be called.

But if its too complicated then its too complicated.